packages/editor/src/components/provider/use-block-editor-settings.js

```diff
@@ -61,11 +62,41 @@ function useBlockEditorSettings( settings, hasTemplate ) {
 			),
 			hasResolvedLocalSiteData: hasFinishedResolvingSiteData,
 			baseUrl: siteData?.url || '',
+			userCanCreatePages: canUser( 'create', 'pages' ),
 		};
 	}, [] );
 
 	const { undo } = useDispatch( editorStore );
 
+	const { saveEntityRecord } = useDispatch( coreStore );
+
+	/**
+	 * Creates a Post entity.
+	 * This is utilised by the Link UI to allow for on-the-fly creation of Posts/Pages.
+	 *
+	 * @param {string} postType the post type of the "post" to be created.
+	 * @param {Object} options  parameters for the post being created. These mirror those used on 3rd param of saveEntityRecord.
+	 * @return {Object} the post type object that was created.
+	 */
+	const createEntity = ( postType, options ) => {
+		// The function is intentionally left open for extension in the future.
```

Comment:
Are we going to need these extensions? It is an experimental function so there is no API lock-in. I think we could add a minimal required API for now and extend it if and when there's a need for that.

Reply:
Ok sure. That's fair enough. Let me make those changes.

Reply:
✅ Changes made.